### PR TITLE
add fused rotary pos emb for qwen

### PR DESCRIPTION
**Pull request by qiuxin2012**
## Description

add fused rotary pos emb for qwen

### 1. Why the change?

optimze qwen's performance.

### CODE REVIEW

**Comment by qiuxin2012** [comment]
Logits diff result if xmx is disabled:
```
optimize model=False logits:
tensor([[[ 0.0691, -0.7929, -2.3262,  ...,  1.1646,  1.1635,  1.1626],
         [ 3.4622,  6.6244,  1.2377,  ..., -5.8193, -5.8102, -5.8098],
         [ 2.3795,  3.9117, -0.1518,  ..., -5.9223, -5.9068, -5.9131],
         ...,
         [ 3.3385,  1.0730, -0.1086,  ..., -5.5679, -5.5583, -5.5563],
         [ 4.1175,  2.0082,  0.4749,  ..., -6.1306, -6.1060, -6.1180],
         [10.1808,  3.9954, -0.0520,  ..., -6.7100, -6.7140, -6.7111]]],
       device='xpu:0')
optimize model=True logits:
tensor([[[ 0.0691, -0.7929, -2.3262,  ...,  1.1646,  1.1635,  1.1626],
         [ 3.4622,  6.6244,  1.2377,  ..., -5.8193, -5.8102, -5.8098],
         [ 2.3795,  3.9117, -0.1518,  ..., -5.9223, -5.9068, -5.9131],
         ...,
         [ 3.3385,  1.0730, -0.1086,  ..., -5.5679, -5.5583, -5.5563],
         [ 4.1175,  2.0082,  0.4749,  ..., -6.1306, -6.1060, -6.1180],
         [10.1808,  3.9954, -0.0520,  ..., -6.7100, -6.7140, -6.7111]]],
       device='xpu:0')
compared different > 1e-5, percentage is 2.3036015164279698e-05
```

**Review by MeouSker77 — APPROVED** [review]

**Comment by qiuxin2012** [comment]
If enable xmx, logits is still under control.
```
tensor([[[ 0.0693, -0.7930, -2.3262,  ...,  1.1641,  1.1631,  1.1621],
         [ 3.4668,  6.6250,  1.2402,  ..., -5.8203, -5.8164, -5.8125],
         [ 2.3809,  3.9141, -0.1504,  ..., -5.9219, -5.9062, -5.9141],
         ...,
         [ 3.3379,  1.0742, -0.1069,  ..., -5.5664, -5.5547, -5.5547],
         [ 4.1133,  2.0078,  0.4785,  ..., -6.1289, -6.1055, -6.1172],
         [10.1719,  3.9902, -0.0576,  ..., -6.7109, -6.7109, -6.7109]]],
       device='xpu:0')
tensor([[[ 0.0693, -0.7910, -2.3262,  ...,  1.1641,  1.1631,  1.1611],
         [ 3.4609,  6.6211,  1.2393,  ..., -5.8164, -5.8086, -5.8086],
         [ 2.3848,  3.9180, -0.1465,  ..., -5.9219, -5.9062, -5.9141],
         ...,
         [ 3.3398,  1.0762, -0.1084,  ..., -5.5742, -5.5547, -5.5586],
         [ 4.1172,  2.0137,  0.4766,  ..., -6.1328, -6.1094, -6.1211],
         [10.1719,  3.9902, -0.0564,  ..., -6.7109, -6.7109, -6.7109]]],
       device='xpu:0')
compared different > 1e-2, percentage is 5.089862398202752e-05
```

**Review by rnwang04 — APPROVED** [review]
